src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -390,6 +453,33 @@ public JSONObject getSlaveAttributeForLabel(String labelName) {
     return null;
   }
 
+  protected Object readResolve() {
```

Comment:
This is a common Jenkins plugin practice to retain backward compatibility, but at the same time migrate the old data to the new data structure. See https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility for details. I will add comments on the implementation details.